components/ws-manager/pkg/manager/create.go

```diff
@@ -212,6 +213,38 @@ func mergeProbe(dst, src reflect.Value) (err error) {
 	return nil
 }
 
+func (m *Manager) createPVCForWorkspacePod(startContext *startWorkspaceContext) (*corev1.PersistentVolumeClaim, error) {
+	req := startContext.Request
+	var prefix string
+	switch req.Type {
+	case api.WorkspaceType_PREBUILD:
+		prefix = "prebuild"
+	case api.WorkspaceType_PROBE:
+		prefix = "probe"
+	case api.WorkspaceType_IMAGEBUILD:
+		prefix = "imagebuild"
+	default:
+		prefix = "ws"
+	}
+	storageClassName := m.Config.Container.PVC.StorageClass
+	return &corev1.PersistentVolumeClaim{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      fmt.Sprintf("%s-%s", prefix, req.Id),
+			Namespace: m.Config.Namespace,
+		},
+		Spec: corev1.PersistentVolumeClaimSpec{
+			AccessModes:      []corev1.PersistentVolumeAccessMode{corev1.ReadWriteOnce},
+			StorageClassName: &storageClassName,
+			Resources: corev1.ResourceRequirements{
+				Requests: corev1.ResourceList{
+					// todo: fix pvc size, should come from ws manager config now
+					corev1.ResourceName(corev1.ResourceStorage): resource.MustParse(m.Config.Container.PVC.Size),
```

Comment:
`MustParse` would panic. If the `PVC.Size` type were `resource.Quantity` already this parse operation would not be necessary.

`resource.Quantity` implements `UnmarshalJSON`.